Dockerfile

```diff
@@ -1,39 +1,33 @@
-FROM opensuse/leap AS elixir-build
+FROM registry.opensuse.org/opensuse/leap:15.4 AS system
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
-RUN zypper -n addrepo https://download.opensuse.org/repositories/devel:/languages:/erlang/SLE_15_SP3/devel:languages:erlang.repo
-RUN zypper -n --gpg-auto-import-keys ref -s
-RUN zypper -n in elixir
-COPY . /build
-WORKDIR /build
 ENV MIX_ENV=prod
-RUN mix local.rebar --force \
-    && mix local.hex --force \
-    && mix deps.get
+RUN zypper -n addrepo https://download.opensuse.org/repositories/devel:/languages:/erlang/15.4/devel:languages:erlang.repo &&\
+    zypper -n --gpg-auto-import-keys ref -s &&\
+    zypper -n in elixir nodejs16 npm16
+WORKDIR /app
+RUN mix local.rebar --force &&\
+    mix local.hex --force
 
-FROM registry.suse.com/bci/nodejs:16 AS assets-build
-COPY --from=elixir-build /build /build
-WORKDIR /build/assets
-RUN npm install
-RUN npm run tailwind:build
-RUN npm run build
+FROM system AS deps
+COPY ./mix.* /app/
+COPY assets/package*.json /app/assets/
+RUN mix deps.all
 
-FROM elixir-build AS release
-COPY --from=assets-build /build /build
-WORKDIR /build
-ENV MIX_ENV=prod
-RUN mix phx.digest
-RUN mix release
+FROM deps AS full
```

Comment:
`release` was a good name :sweat_smile:

Reply:
The rationale about `full` and `minimal` is that both could be used as final images. One only has the static binary, the other has the full development toolkit.
For the demo environment, for instance, we will probably want to use the `full` variant, so that we can tinker with `mix` tasks and the code on a live installation.

Reply:
nitpick: i also like release better too, i find the naming proposed a bit confusing

Reply:
@stefanotorresi gotcha, thanks for clarifing

Reply:
@stefanotorresi well your rationale was good but I didn't catch it at a first glance, so... :sweat_smile:

Reply:
let me copy the `ENTRYPOINT` and `EXPOSE` statements in there as well, so that we make this more apparent.